### ISSUE TITLE
What happens when playing pure audio fails? We only did audio push-pull streaming, but we ran into problems playing it back.

### ISSUE THREAD

**Issue by Balopy** [shogo4405/HaishinKit.swift#1354]
### Describe the bug
```
2023-12-14 00:52:11.427424-0800 RTMPPushStreamDemo[79417:8597476] [as] ATAudioSessionPropertyManager.mm:363   FAILED to set property 1684431725 on AVAudioSession with error Error Domain=NSOSStatusErrorDomain Code=-50 "(null)"
2023-12-14 00:52:11.819220-0800 RTMPPushStreamDemo[79417:8597476] Metal API Validation Enabled
2023-14-12 00:52:12.069 [Info] [com.haishinkit.Exsample.iOS] [IngestViewController.swift:52] viewWillAppear(_:) > viewWillAppear
2023-14-12 00:52:12.139 [Info] [com.haishinkit.Exsample.iOS] [IngestViewController.swift:174] didRouteChangeNotification(_:) > name = AVAudioSessionRouteChangeNotification, object = Optional(<AVAudioSession: 0x280f80760>), userInfo = Optional([AnyHashable("AVAudioSessionRouteChangePreviousRouteKey"): <AVAudioSessionRouteDescription: 0x280f90090, 
inputs = (
); 
outputs = (
    "<AVAudioSessionPortDescription: 0x280f900c0, type = Speaker; name = \U626c\U58f0\U5668; UID = Speaker; selectedDataSource = (null)>"
)>, AnyHashable("AVAudioSessionRouteChangeReasonKey"): 3])
2023-14-12 00:52:12.216 [Info] [com.haishinkit.Exsample.iOS] [IngestViewController.swift:174] didRouteChangeNotification(_:) > name = AVAudioSessionRouteChangeNotification, object = Optional(<AVAudioSession: 0x280f80760>), userInfo = Optional([AnyHashable("AVAudioSessionRouteChangeReasonKey"): 3, AnyHashable("AVAudioSessionRouteChangePreviousRouteKey"): <AVAudioSessionRouteDescription: 0x280f900a0, 
inputs = (
    "<AVAudioSessionPortDescription: 0x280f90070, type = MicrophoneBuiltIn; name = iPhone \U9ea6\U514b\U98ce; UID = Built-In Microphone; selectedDataSource = \U4e0b>"
); 
outputs = (
    "<AVAudioSessionPortDescription: 0x280f8da30, type = Receiver; name = \U63a5\U6536\U5668; UID = Built-In Receiver; selectedDataSource = (null)>"
)>])
2023-14-12 00:52:12.228 [Info] [com.haishinkit.Exsample.iOS] [IngestViewController.swift:174] didRouteChangeNotification(_:) > name = AVAudioSessionRouteChangeNotification, object = Optional(<AVAudioSession: 0x280f80760>), userInfo = Optional([AnyHashable("AVAudioSessionRouteChangeReasonKey"): 3, AnyHashable("AVAudioSessionRouteChangePreviousRouteKey"): <AVAudioSessionRouteDescription: 0x280f81880, 
inputs = (
    "<AVAudioSessionPortDescription: 0x280f81820, type = MicrophoneBuiltIn; name = iPhone \U9ea6\U514b\U98ce; UID = Built-In Microphone; selectedDataSource = (null)>"
); 
outputs = (
    "<AVAudioSessionPortDescription: 0x280f818f0, type = Receiver; name = \U63a5\U6536\U5668; UID = Built-In Receiver; selectedDataSource = (null)>"
)>])
2023-14-12 00:52:16.472 [Info] [com.haishinkit.Exsample.iOS] [IngestViewController.swift:71] viewWillDisappear(_:) > viewWillDisappear
2023-14-12 00:52:18.530 [Info] [com.haishinkit.HaishinKit] [RTMPNWSocket.swift:168] stateDidChange(to:) > Connection is ready.
2023-14-12 00:52:18.548 [Info] [com.haishinkit.HaishinKit] [RTMPNWSocket.swift:159] viabilityDidChange(to:) > Connection viability changed to true
2023-14-12 00:52:19.153 [Info] [com.haishinkit.Exsample.iOS] [PreferenceViewController.swift:79] rtmpStatusHandler(_:) > NetConnection.Connect.Success
2023-14-12 00:52:19.614 [Info] [com.haishinkit.HaishinKit] [IOMixer.swift:309] mediaLink(_:didBufferingChanged:) > true
2023-14-12 00:52:19.865 [Info] [com.haishinkit.Exsample.iOS] [PreferenceViewController.swift:79] rtmpStatusHandler(_:) > NetStream.Play.Reset
2023-14-12 00:52:19.866 [Info] [com.haishinkit.Exsample.iOS] [PreferenceViewController.swift:79] rtmpStatusHandler(_:) > NetStream.Play.Start
2023-12-14 00:52:19.974683-0800 RTMPPushStreamDemo[79417:8597761] [avae]            AVAEInternal.h:130   [AUInterface.mm:539:SetFormat: ([[busArray objectAtIndexedSubscript:(NSUInteger)element] setFormat:format error:&nsErr])] returned false, error Error Domain=NSOSStatusErrorDomain Code=-10868 "(null)"
2023-14-12 00:52:19.977 [Warn] [com.haishinkit.HaishinKit] [IOMixer.swift:68] audioFormat > [[busArray objectAtIndexedSubscript:(NSUInteger)element] setFormat:format error:&nsErr]: returned false, error Error Domain=NSOSStatusErrorDomain Code=-10868 "(null)"
2023-14-12 00:52:20.144 [Warn] [com.haishinkit.HaishinKit] [MediaLink.swift:89] enqueueAudio(_:) > player started when in a disconnected state
2023-14-12 00:52:20.145 [Warn] [com.haishinkit.HaishinKit] [MediaLink.swift:89] enqueueAudio(_:) > player started when in a disconnected state
2023-14-12 00:52:20.146 [Warn] [com.haishinkit.HaishinKit] [MediaLink.swift:89] enqueueAudio(_:) > player started when in a disconnected state
2023-14-12 00:52:20.147 [Warn] [com.haishinkit.HaishinKit] [MediaLink.swift:89] enqueueAudio(_:) > player started when in a disconnected state
```
URI： rtmp://test-obradio.****.com:8888/obradio/1234，  streamName：1234

Does our code support audio-only playback? Is playing audio files supported?

Such as: http://test-obradio.****.com/obradio/1234.flv

### To Reproduce

no

### Expected behavior

no

### Version

1.6.2

**Comment by shogo4405** [comment]
I'm not sure due to the insufficient details in the issue. I recommend posting a question on https://stackoverflow.com/ for a more comprehensive answer.

**Comment by Balopy** [comment]
My question is, does it support audio-only playback? Is there a demo for that? The demo on github did not play successfully, 
the following error
2023-14-12 00:52:19.866 [Info] [com.haishinkit.Exsample.iOS] [PreferenceViewController.swift:79] rtmpStatusHandler(_:) > NetStream.Play.Start
2023-12-14 00:52:19.974683-0800 RTMPPushStreamDemo[79417:8597761] [avae]            AVAEInternal.h:130   [AUInterface.mm:539:SetFormat: ([[busArray objectAtIndexedSubscript:(NSUInteger)element] setFormat:format error:&nsErr])] returned false, error Error Domain=NSOSStatusErrorDomain Code=-10868 "(null)"
2023-14-12 00:52:19.977 [Warn] [com.haishinkit.HaishinKit] [IOMixer.swift:68] audioFormat > [[busArray objectAtIndexedSubscript:(NSUInteger)element] setFormat:format error:&nsErr]: returned false, error Error Domain=NSOSStatusErrorDomain Code=-10868 "(null)"
2023-14-12 00:52:20.144 [Warn] [com.haishinkit.HaishinKit] [MediaLink.swift:89] enqueueAudio(_:) > player started when in a disconnected state

**Comment by shogo4405** [comment]
Yes, I have confirmed that audio-only playback is possible in version 1.7.2. However, it is limited to AAC 2 channels.

**Comment by Balopy** [comment]
That's great, but is there a demo where you can play audio? Because I'm using xcode14.2, I can't upgrade xcode15

**Comment by shogo4405** [comment]
Yes, I provide an Example app. You can check it in the iOS app in the Example directory.
https://github.com/shogo4405/HaishinKit.swift/tree/main/Example
Also You can download from tags. https://github.com/shogo4405/HaishinKit.swift/releases 

There is a bug from a certain version of 1.5.x that allows playback of only 1-channel AAC audio. I fixed this issue before the release of 1.7.2, so it might be related to that problem. There are no plans to backport it for Xcode 14.

**Comment by Balopy** [comment]
Hello, When I was using demo, I encountered such a problem, how to solve it?



➜  HaishinKit.swift-1.6.1 ls
Cartfile             HaishinKit.podspec   Platforms            Vendor
Cartfile.resolved    HaishinKit.xcodeproj README.md            docs
Examples             LICENSE.md           SRTHaishinKit        fastlane
Gemfile              Package.resolved     Sources
Gemfile.lock         Package.swift        Tests
➜  HaishinKit.swift-1.6.1 carthage bootstrap --platform iOS,macOS,tvOS --use-xcframeworks

*** Checking out Logboard at "2.4.1"
*** Cloning Logboard
*** xcodebuild output can be found in /var/folders/cd/wl2kwqvx4bz6w96h2bqvtgvh0000gn/T/carthage-xcodebuild.vPVqH5.log
*** Skipped downloading Logboard binary due to the error:
	"API rate limit exceeded for 192.169.96.216. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)"
*** Building scheme "Logboard" in Logboard.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/wangchunlong/Downloads/HaishinKit.swift-1.6.1/Carthage/Checkouts/Logboard/Logboard.xcworkspace -scheme Logboard -configuration Release -derivedDataPath /Users/wangchunlong/Library/Caches/org.carthage.CarthageKit/DerivedData/14.2_14C18/Logboard/2.4.1 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive VALIDATE_WORKSPACE=NO -archivePath /var/folders/cd/wl2kwqvx4bz6w96h2bqvtgvh0000gn/T/Logboard SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/wangchunlong/Downloads/HaishinKit.swift-1.6.1/Carthage/Checkouts/Logboard)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/cd/wl2kwqvx4bz6w96h2bqvtgvh0000gn/T/carthage-xcodebuild.vPVqH5.log
➜  HaishinKit.swift-1.6.1
